### PR TITLE
Use docker official redis images

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:

It fixes the regression introduced in https://github.com/kubernetes/kubernetes/pull/79388
breaking the CI for non-amd64 platforms when IPv6 support was added

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This PR replaces current redis images used in e2e tests by the docker official images that have multi-platform support, reducing the cost of supporting custom images for e2e testing

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func initImageConfigs() map[int]Config {
 	configs[Perl] = Config{dockerLibraryRegistry, "perl", "5.26"}
```
Comment (mkumatag) [inline test/utils/image/manifest.go]: see if no one else using GBRedisSlaveImage then can we just remove an entry from here as well?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: test/utils/image/manifest.go (hunk 1 above)
Comment (aojea) [reply test/utils/image/manifest.go]: `grep -r GBRedis test/` doesn´t return anything, seem is safe to remove it

**Review by BenTheElder — APPROVED** [review]
/approve

**Pull request merged**